src/jaxsim/simulation/integrators.py

```diff
@@ -294,51 +234,13 @@ def body_fun(carry: Carry, xs: None) -> Tuple[Carry, None]:
 
         return carry, None
 
-    # Integrate over the given horizon
-    (x_tf, _), _ = jax.lax.scan(
-        f=body_fun, init=carry_init, xs=None, length=num_sub_steps
-    )
-
-    # Compute the aux dictionary at t0
-    _, aux_t0 = dx_dt(x0, t0)
-
-    return x_tf, aux_t0
-
-
-def odeint_euler_semi_implicit_manifold_one_step(
-    dx_dt: StateDerivativeCallable,
-    x0: ODEState,
-    t0: Time,
-    tf: Time,
-    num_sub_steps: int = 1,
-) -> Tuple[ODEState, Dict[str, Any]]:
-    """
-    Semi-implicit Euler integrator with quaternion integration on SO(3).
-
-    Args:
-        dx_dt: Callable that computes the state derivative.
-        x0: Initial state as ODEState object.
-        t0: Initial time.
-        tf: Final time.
-        num_sub_steps: Number of sub-steps to break the integration into.
-
-    Returns:
-        A tuple having as first element the final state as ODEState object,
-        and as second element a dictionary including auxiliary data at t0.
-    """
-
-    # Compute the sub-step size.
-    # We break dt in configurable sub-steps.
-    dt = tf - t0
-    sub_step_dt = dt / num_sub_steps
-
-    # Integrate the quaternion on its manifold using the new angular velocity
+    def semi_implicit_euler_manifold_body_fun(
```

Comment:
Can we re-use the variant with no integration on the manifold also here? I don't mind if we need to do the extra integration of the corresponding 4 elements to later override them.

Reply:
Addressed in https://github.com/ami-iit/jaxsim/pull/82/commits/1d2c91cac2959f1f0a76d7330db6c05682a9a8a7